### PR TITLE
Add support for detecting content-type charset

### PR DESCRIPTION
**Pull request by er1c**

### CODE REVIEW

**Comment by er1c** [comment]
PR updated with "negative" test

File: src/test/scala/fm/http/TestClientAndServer.scala [hunk 1]
```diff
@@ -43,11 +43,19 @@ object TestClientAndServer {
...
+  private val Latin1Header: Headers = Headers(("Content-Type", "text/html; charset=ISO-8859-1"))
+
   protected val unwrappedHandler: PartialFunction[Request, Response] = {
     case GET(simple"/${INT(code)}") => Response(Status(code), Status(code).msg)
     case GET("/data_one_mb")        => Response.Ok(makeLinkedHttpContent(OneMB))
     case GET("/data_hundred_mb")    => Response.Ok(makeLinkedHttpContent(OneMB * 100))
+
+    case GET("/utf-8")                   => Response.Ok(UTF8Header, Unpooled.copiedBuffer("£", CharsetUtil.UTF_8))
+    case GET("/latin1")                  => Response.Ok(Latin1Header, Unpooled.copiedBuffer(Array(163).map { _.toByte }))  // 163 is latin1 £ character
+    case GET("/default-latin1")          => Response.Ok(Headers.empty, Unpooled.copiedBuffer(Array(163).map { _.toByte })) // 163 is latin1 £ character
```
Comment (tpunder) [inline src/test/scala/fm/http/TestClientAndServer.scala]: Any reason you didn't use:

```
Unpooled.copiedBuffer("£", CharsetUtil.ISO_8859_1)
```

?

File: src/test/scala/fm/http/TestClientAndServer.scala [hunk 2]
```diff
@@ -207,4 +215,31 @@ final class TestClientAndServer extends FunSuite with Matchers with BeforeAndAft
 //      res should equal (true)
 //    }
 //  }
+
+  /*
+  Certain characters like £ are two bytes in UTF-8 and one byte in Latin1
+  Test support for reading explicit latin1/utf-8/etc response body and "mapping" to native java utf8 string
```
Comment (tpunder) [inline src/test/scala/fm/http/TestClientAndServer.scala]: > native java utf8 string

That's not an accurate comment.  Native Java strings are not UTF-8

**Comment by er1c** [comment]
PR updated

**Pull request merged**